### PR TITLE
UUID instead of Integer for ID

### PR DESCRIPTION
**Pull request by yermaalexx**
changed type of ID for entities and DTO: from Integer to UUID

### CODE REVIEW

File: src/main/java/com/alexx/employeesAndDepartments/controllers/DepartmentController.java [hunk 1]
```diff
@@ -37,28 +38,25 @@ public ResponseEntity<DepartmentDTO> editDepartment(@RequestBody DepartmentDTO d
...
     }
 
     @GetMapping("/departments/{departmentId}")
     @Operation(summary = "Get department", description = "Get an existing department with its list of employees, ID of department in path required")
     public ResponseEntity<DepartmentDTO> getDepartment(@PathVariable("departmentId")
-                                                           @Min(value = 1, message = "ID of department must be positive.")
-                                                           @Parameter(description = "ID of the existing department to get")
-                                                           Integer id) {
+                                                       @Parameter(description = "ID of the existing department to get")
+                                                       UUID id) {
```
Comment (yermakovsa) [inline src/main/java/com/alexx/employeesAndDepartments/controllers/DepartmentController.java]: Please use String as a parameter in requests and then internally convert it to UUID (UUID.fromString(...)

File: src/main/java/com/alexx/employeesAndDepartments/models/DepartmentDTO.java [hunk 2]
```diff
@@ -8,13 +8,14 @@
...
+import java.util.UUID;
 
 @Schema(description = "Department description")
 public class DepartmentDTO {
 
   @NotNull(groups = Marker.OnUpdate.class, message = "ID of department must not be empty.")
-  @Schema(description = "Automatically generated department ID", example = "2")
-  private Integer id;
+  @Schema(description = "Automatically generated department ID", example = "86c6ff27-a386-4479-8cec-cfde91cb9474")
+  private UUID id;
```
Comment (yermakovsa) [inline src/main/java/com/alexx/employeesAndDepartments/models/DepartmentDTO.java]: Please use string here

Comment (yermakovsa) [reply src/main/java/com/alexx/employeesAndDepartments/models/DepartmentDTO.java]: Use UUID only in entities

File: src/main/java/com/alexx/employeesAndDepartments/controllers/DepartmentController.java (hunk 1 above)
Comment (yermaalexx) [reply src/main/java/com/alexx/employeesAndDepartments/controllers/DepartmentController.java]: got it, will fix

File: src/main/java/com/alexx/employeesAndDepartments/models/DepartmentDTO.java (hunk 2 above)
Comment (yermaalexx) [reply src/main/java/com/alexx/employeesAndDepartments/models/DepartmentDTO.java]: got it, will fix

**Review by yermakovsa — APPROVED** [review]

**Pull request merged**